Code/ForceField/Wrap/ForceField.cpp

```diff
@@ -123,6 +123,61 @@ PyObject *ForceFieldGetExtraPointLoc(PyForceField *self, unsigned int idx) {
   return res;
 }
 
+double PyForceField::calcEnergyWithPos(const python::object &pos) {
+  PRECONDITION(this->field, "no force field");
+  if (pos != python::object()) {
+    int s = this->field->dimension() * this->field->numPoints();
+    int numElements = python::extract<int>(pos.attr("__len__")());
+    PRECONDITION(s == numElements,
+      "The Python container must have length equal to Dimension() * NumPoints()");
+    std::vector<double> c(s);
+    for (int i = 0; i < s; ++i)
+      c[i] = python::extract<double>(pos[i]);
+    return this->field->calcEnergy(c.data());
+  }
+  else
+    return this->field->calcEnergy();
+}
+
+PyObject *PyForceField::positions() {
+  PRECONDITION(this->field, "no force field");
+  int s = this->field->dimension() * this->field->numPoints();
+  PyObject *coordTuple = PyTuple_New(s);
+  const RDGeom::PointPtrVect &p = this->field->positions();
+  int i = 0;
+  PyObject *coordItem;
+  for (RDGeom::PointPtrVect::const_iterator it = p.begin(); it != p.end(); ++it) {
```

Comment:
Since this is on the modern c++ branch, this use a range for loop. Something like (not tested, which is dangerous since I'm not completely fluent here yet):
```
for(auto pptr : p){
  for(size_t j=0;j<3;++j){
    coordItem = PyFloat_FromDouble((*pptr)[j]);
    ....
  }
}